### ISSUE TITLE
[Feature Request] Bring back vanilla Creative searching as an optional thing, or add redirects when attempting to use vanilla search

### ISSUE THREAD

**Issue by Roadhog360** [GTNewHorizons/NotEnoughItems#491]
Yes I know it is slow. It could be optimized admittedly. But it's faster for me because I already have muscle memory of opening it, since when I open my inventory in a Creative world after starting a game, Creative search would be already open and available for me to type, so I'm used to just pressing E and instantly typing what I want to grab. So it's annoying when I do that and get halfway through typing only to realize that was wasted, then I have to click over to my survival inventory to make the search bar appear, then click the search bar which is much slower and many more clicks. While in a production environment this isn't an issue due to not constantly restarting, in dev I'm constantly doing this by accident since restarting the game places your inventory back in the Creative search tab by default. And sometimes out of pure habit when rearranging my inventory I'll want another item and instantly go into that search tab without thinking.

Here are several ideas I have to reduce the click count on the UI and preserve muscle memory for people like me who have been doing this for years, all of which should be elegant solutions that preserve muscle memory, whilst would also provide convenient shortcuts for people who aren't experiencing my issue. Win-win?
1. When in the Creative inventory, or specifically the search tab, make it so NEI isn't hidden. Attempting to type in the search text field will then redirect you to to the NEI search which would be visible here. So when I open my inventory and start typing, it'll just select the NEI text box and type there.
2. When the search tab in the Creative menu is opened, it redirects you to the survival inventory and auto-selects the search tab. This would also solve the issue as when the game first opens the Creative search tab by default, since it'd then make it open the survival inventory with the search bar auto selected to type in.
3. Make it a config option to re-enable Creative search. In most cases I use vanilla search, it's in dev and isn't that slow. I usually end up using the NEI search in production since again, production isn't when I'm constantly restarting and being redirected to the vanilla search tab. It could be optimized, but in all honestly it doesn't need to. Since I am primarily making this request with dev env efficiency in mind, most dev envs aren't loaded with enough mods for vanilla search's slowness to be apparent. So it really isn't that slow for my use case.
4. Remove the standard Creative mode, and make Creative+ supersede it. This is the most straightforward approach, and would forcefully break my habit as this would entirely nuke vanilla search, so there's nothing to accidentally trigger muscle memory.

Please let me know what you think.

**Comment by Roadhog360** [comment]
My mistake; this is a Hodgepodge config option. I'd like to amend my suggestion to move that option to GTNH-NEI instead. And I think implementing these suggestions would be useful even with the blocking disabled, so I'll keep it open and see what the contributors think.

**Comment by OvermindDL1** [comment]
Probably a simple mixin for when the search tab is clicked it instead switches to the C+ gui and highlights search would be the most simple, though tying in NEI's search on the vanillas search panel would be the best 'experience' I'd wager...

**Comment by Roadhog360** [comment]
Yeah, I'd be totally down to just put NEI's search logic into the vanilla Creative search menu. I think that should actually be handled by NEI; it makes no sense for Hodgepodge to be the one behind this.

**Comment by joshaup1999** [comment]
Or.. hear me out… disable the vanilla creative mode and switch it to the creative+ mode from nei

**Comment by Roadhog360** [comment]
I'm glad you read 0% of the post, this doesn't actually solve the core issue
